### PR TITLE
Add sentry to Core and SQLAlchemy

### PR DESCRIPTION
**Pull request by not4Pedro**
Sentry succesfully collects data from Core and DB
- [x] add info what sentry monitors
- [ ] document in the docs

### CODE REVIEW

File: src/core/core/__init__.py [hunk 1]
```diff
@@ -19,6 +23,20 @@ def create_app(initial_setup: bool = True):
...
+    sentry_sdk.init(
+        dsn = os.getenv("SENTRY_DSN"),
+        # Set traces_sample_rate to 1.0 to capture 100%
+        # of transactions for tracing.
+        traces_sample_rate = os.getenv("SENTRY_TRACES_SAMPLE_RATE", 1.0),
+        # Set profiles_sample_rate to 1.0 to profile 100%
+        # of sampled transactions.
+        # We recommend adjusting this value in production.
+        profiles_sample_rate=os.getenv("SENTRY_PROFILES_SAMPLE_RATE", 1.0)
+    )
```
[the remark this answers is not in the document]
Comment (not4Pedro) [reply src/core/core/__init__.py]: the default return value of `os.getenv` is None, Sentry handles this by not sending any requests to Sentry

File: src/core/core/__init__.py [hunk 2]
```diff
@@ -19,6 +24,20 @@ def create_app(initial_setup: bool = True):
...
+        # Set traces_sample_rate to 1.0 to capture 100%
+        # of transactions for tracing.
+        traces_sample_rate = float(os.getenv("SENTRY_TRACES_SAMPLE_RATE", 1.0)),
+        # Set profiles_sample_rate to 1.0 to profile 100%
+        # of sampled transactions.
+        # We recommend adjusting this value in production.
+        profiles_sample_rate=float(os.getenv("SENTRY_PROFILES_SAMPLE_RATE", 1.0))
+    )
+
+
```
[the remark this answers is not in the document]
Comment (not4Pedro) [reply src/core/core/__init__.py:40]: @b3n4kh okay, as per example, sentry starts before that app, but apparently that's not mandatory. I will do

File: src/core/core/__init__.py [hunk 3]
```diff
@@ -25,3 +26,4 @@ def initialize_managers(app: Flask, initial_setup: bool = False):
     api_manager.initialize(app)
     queue_manager.initialize(app, initial_setup)
     data_manager.initialize(initial_setup)
+    sentry_manager.initialize(app)
```
Comment (b3n4kh) [inline src/core/core/__init__.py:29]: should be one of the first things, to get logs if anything during initialization fails

File: src/core/core/managers/sentry_manager.py [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+
+def initialize(app):
+    sentry_sdk.init(
+        dsn=app.config.get("SENTRY_DSN"),
+        # Set traces_sample_rate to 1.0 to capture 100%
+        # of transactions for tracing.
+        traces_sample_rate=app.config.get("SENTRY_TRACES_SAMPLE_RATE"),
+        # Set profiles_sample_rate to 1.0 to profile 100%
+        # of sampled transactions.
+        # We recommend adjusting this value in production.
```
Comment (b3n4kh) [inline src/core/core/managers/sentry_manager.py:12]: Remove copy/pasted comments

**Review by b3n4kh — CHANGES_REQUESTED** [review]

File: docker/compose.yml [hunk 5]
```diff
@@ -15,7 +15,7 @@ services:
       NGINX_CONNECTIONS: "16"
       TARANIS_CORE_UPSTREAM: "${TARANIS_CORE_UPSTREAM:-core:8080}"
       TARANIS_SSE_UPSTREAM: "${TARANIS_SSE_UPSTREAM:-sse:8088}"
-      TARANIS_SENTRY_DSN: "${TARANIS_SENTRY_DSN:-}"
+      TARANIS_SENTRY_DSN: "${TARANIS_GUI_SENTRY_DSN:-}"
```
Comment (b3n4kh) [inline docker/compose.yml:18]: change both to: `TARANIS_GUI_SENTRY_DSN`

File: docker/compose.yml [hunk 6]
```diff
@@ -38,17 +38,14 @@ services:
...
-      DB_POOL_TIMEOUT: 5
-
       TARANIS_AUTHENTICATOR: "${TARANIS_AUTHENTICATOR:-database}"
 
       QUEUE_BROKER_HOST: "rabbitmq"
       QUEUE_BROKER_USER: "${RABBITMQ_USER:-taranis}"
       QUEUE_BROKER_PASSWORD: "${RABBITMQ_PASSWORD:-taranis}"
 
       GRANIAN_PORT: "${TARANIS_CORE_LISTEN_PORT:-8080}"
+      SENTRY_DSN: "${TARANIS_CORE_SENTRY_DSN:-}"
```
Comment (b3n4kh) [inline docker/compose.yml:48]: change both to: `TARANIS_CORE_SENTRY_DSN`

**Review by b3n4kh — CHANGES_REQUESTED** [review]

**Pull request merged**